.storybook/test-runner.ts

```diff
@@ -0,0 +1,15 @@
+// TODO: Convert this to ESM once the bug is resolved
+// https://github.com/storybookjs/test-runner/issues/293
+
+const config = {
+  async postRender(page, _context) {
+    const elem = await page.locator('[data-test-id="story-root"]');
+    if (!elem) {
```

Comment:
Hm I was actually talking about making the test fail instead of throwing an error 🙈  Like first assert that the element exists :)

Reply:
```tsx
const rootElementId = '[data-test-id="story-root"]';

const rootElement = await page.locator(rootElementId);

expect(rootElement).toBeDefined();

if (rootElement) {
  const content = await rootElement.innerHTML();

  expect(html).toMatchSnapshot();
}
```

Reply:
Also not sure if this `await rootElement.innerHTML()` is part of this API you mentioned, but usually on the DOM API, `rootElement.innerHTML` should be sync and a property 🤔

Reply:
Sounds good, my bad!

Reply:
Although the `if` shouldn't be needed since we're already checking if `rootElement` is defined.

Reply:
@ovflowd It's part of the api of `page` that playwright provides. I used innerHTML() as well and was shocked it didn't work and had to add the `await` later.

Reply:
The `if` is needed as the assertion failing doesn't stop the test from running :)

Reply:
I believe `jest` stops execution when the first assertion fails.
Here's a simple example
https://stackblitz.com/edit/jest-example-josuki?file=sum.test.js

Reply:
I'm not sure that's how it works (by personal experience). If you could actually test it is doing what you expect by manually making it fail, let me know.

Reply:
In the above case, seems like `toBeDefined` is doing nothing because the `elem` will always be defined even if it doesn't exist
<img width="485" alt="image" src="https://user-images.githubusercontent.com/10580591/235350254-13f04d39-b10a-4cd9-a507-a3560e3fcfa3.png">
which is why the test continues.

Reply:
And `content` is returned as `undefined` when the `elem` doesn't exist without causing any errors
<img width="351" alt="image" src="https://user-images.githubusercontent.com/10580591/235350341-2d8b7386-a47b-47a4-ba0a-292c3decb150.png">

Reply:
Then we don't need the if checks

But let's add a toBeDefined after the await of the innerHTML

Reply:
Updated!